kinesis/kinesis.go

```diff
@@ -194,36 +197,120 @@ func (outputPlugin *OutputPlugin) AddRecord(record map[interface{}]interface{},
 		return fluentbit.FLB_OK
 	}
 
-	newRecordSize := len(data) + len(partitionKey)
-
-	if len(outputPlugin.records) == maximumRecordsPerPut || (outputPlugin.dataLength+newRecordSize) > maximumPutRecordBatchSize {
-		retCode, err := outputPlugin.sendCurrentBatch()
-		if err != nil {
-			logrus.Errorf("[kinesis %d] %v\n", outputPlugin.PluginID, err)
-		}
-		if retCode != fluentbit.FLB_OK {
-			return retCode
-		}
-	}
-
-	outputPlugin.records = append(outputPlugin.records, &kinesis.PutRecordsRequestEntry{
+	*records = append(*records, &kinesis.PutRecordsRequestEntry{
 		Data:         data,
 		PartitionKey: aws.String(partitionKey),
 	})
-	outputPlugin.dataLength += newRecordSize
 	return fluentbit.FLB_OK
 }
 
 // Flush sends the current buffer of log records
 // Returns FLB_OK, FLB_RETRY, FLB_ERROR
-func (outputPlugin *OutputPlugin) Flush() int {
-	retCode, err := outputPlugin.sendCurrentBatch()
+func (outputPlugin *OutputPlugin) Flush(records *[]*kinesis.PutRecordsRequestEntry) int {
```

Comment:
nit: Is the parameter type `*[]*kinesis.PutRecordRequestEntry` in correct format? Ignore my knowledge with golang syntax. I think there is an extra "*".